Feature/new version management script

loremattei opened the pull request:
## Description
Adds a script named manage-version.sh that supports some configuration updates needed on version changes.

The script  accepts two commands: create-branch and update-branch.

The create command:
1. checks out the develop branch, pulls any update and creates a new release branch named release/X.Y and checks out it ;
2. updates the version in the GlotPress related keys of the relevant configuration files;
3. updates the version in the Fastlane configuration file;
4. updates the version in the pList files.

The update command performs the steps 3 and 4 in the current branch.

In order to run, the script needs at least one version string (in the X.Y.Z.build format). If the version string has less than 4 parts, they are supposed to be 0 (i.e. 9.3 => 9.3.0.0).

The script accepts a second version string in the X.Y.Z.date format that will be used to update the pList files related to the internal builds. When the second string is not provided, it is builded from the first one and using today date. 

## To test:
From the root directory of your Wordpress checkout, run:
`# cd Scripts`

### Test 1:
`# ./manage-version.sh create <version-string>`
ex: ./manage-version.sh create 9.3.1.0

Verify that:
 - the script runs without errors;
 - the release/X.Y branch is created;
 - the download_metadata.swift and AppStoreStrings.po files are correctly updated;
 - the Scripts/fastlane/Deliverfile is correctly updated;
 - all the 9 .plist files are correctly updated.

### Test 2:
`# ./manage-version.sh update <version-string>`
ex: ./manage-version.sh update 9.4

Verify that:
 - the script runs without errors;
 - the Scripts/fastlane/Deliverfile is correctly updated;
 - all the 9 .plist files are correctly updated.

### Test 3:
`# ./manage-version.sh update <version-string> <internal-build-version-string>`
ex: ./manage-version.sh update 9.4.1.2  9.4.1.20181212

Verify that:
 - the script runs without errors;
 - all the 9 .plist files are correctly updated.

@elibud @diegoreymendez 
fyi @astralbodies

elibud commented:
I really like this. The code is easy to follow (for bash script :P) and it works as advertised, good job.

Here's the results of my tests, let me know what you think about the feedback.

**Create test 1** `./manage-version.sh create 9.3.0` Branch already exists, current version number is `9.3.0.1` 
-> This reverted my version number to `9.3.0`, I would have expected it to fail since it would be going back a version

**Create test 2** `./manage-version.sh create 9.3.0.1` Branch already exists, current version number is `9.3.0.1` 
-> Worked fine

**Create test 3** `./manage-version.sh create 9.4` Branch doesn't exist, current version number is `9.3.0.` -> Worked fine

**Create test 4** `./manage-version.sh create 8.8` Branch doesn't exist, current version number is `9.3.0.` -> It created a new version, again I would've expected it to fail since that's a version older than the current one.

**Update test 1** `./manage-version.sh update 9.4`, this branch current version number is `9.3.0.1` -> I'm not sure executing step 3 makes a lot of sense here, since the `Deliverfile` should be updated already from the `create` execution. At least I can not justify updating the `Deliverfile` and not the `GlotPress` keys but I might be missing something.

**Update test 2** `./manage-version.sh update 9.4.0.3 20180203
Internal and external versions don't match.` -> Got an error: Internal and external versions don't match. :) Awesome :)

**Update test 3** `./manage-version.sh update 9.4.0.3 9.4.0.20180203` -> All good

loremattei commented:
@elibud thank you very much.
I'm updating the script with checks against going back with version (and a new "force" command to skip them in case of necessity).

About the "update test 1": You are right. I think the correct behaviour here could be not updating the Deliverfile in normal cases, while updating both the Deliverfile and the glotPress related ones in case of "force".

loremattei commented:
Hi,
some mods:
- added the checks against going back with versions;
- removed the point 3 from "update" command;
- added a "force" command to skip version checks (to be used to fix errors, mismatches, etc.)

"Create test 1", "Create test 4" and "Update test 1" should be fine now.

@elibud @diegoreymendez
